mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/documentdialog/DocumentDialogFragment.java

```diff
@@ -176,9 +182,24 @@ public void validateInput() throws RequiredFieldException {
 
         //Start Uploading Document
         if (documentAction == getResources().getString(R.string.update_document)) {
+            updatedDocument = document;
+            updatedDocument.setName(documentName);
+            updatedDocument.setDescription(documentDescription);
+            updatedDocument.setFileName(fileChoosen.getName());
+            updatedDocument.setSize(fileChoosen.length());
+
             mDocumentDialogPresenter.updateDocument(entityType, entityId, document.getId(),
                     documentName, documentDescription, fileChoosen);
+
         } else if (documentAction == getResources().getString(R.string.upload_document)) {
+            createdDocument = new Document();
+            createdDocument.setName(documentName);
+            createdDocument.setDescription(documentDescription);
+            createdDocument.setFileName(fileChoosen.getName());
+            createdDocument.setParentEntityType(entityType);
+            createdDocument.setParentEntityId(entityId);
+            createdDocument.setSize(fileChoosen.length());
```

Comment:
Too much code copied, can we abstract out a method for doing something to your created or updated documents.

Some like methodA() that does the common work and methodB() that first calls methodA() and then also does your extra work.

Let me know if this makes sense and it is just an idea to discuss if it could possibly improve the code.

Reply:
Yes, we can do that. In my opinion, it won't provide substantial abstraction but we can discuss this too.

Reply:
I've changed it.